### DISCUSSION TITLE
Constructors from Optional Variants?

### DISCUSSION THREAD

**Discussion by migueldeicaza** [migueldeicaza/SwiftGodot#448] [Ideas]
One idiom I keep running into is extracting values out of a Dictionary (which could be nil) and then having to add extra code to convert it to t a value:

Like this:
```
guard let keyVariant = dict ["key"] else { 
    print ("Not found")
    return 
}
```

then try to convert this to say, a String, or Int or whatever else, with the constructors that take a "Variant":

```
guard let realValue = String (keyVariant) else { 
    print ("Could not convert the variant to a String")
    return 
}
```

I am tempted to add failable constructors that take a "Variant?" rather than just a Variant:

They are already failable, but this would extend it to allow optionals

So I could write

```
guard let realValue = String (dict ["key"]) else { 
   print ("Key not found, or the value is not a string")
   return 
}
```

Are there any downsides I might not be anticipating?

**Comment by shanayyy** [comment]
🌝I think it’s also good to use Optional.flatMap:
```
guard let realValue = dict["key"].flatMap(String.init) else {
    print ("Key not found, or the value is not a string")
    return
}
```
